File-Formats/DocIO/word-to-markdown.md

```diff
@@ -0,0 +1,800 @@
+{% endhighlight %}
+
+{% highlight c# tabtitle="ASP.NET Core" %}
+//Open the file as a Stream.
+using (FileStream docStream = new FileStream("Input.docx", FileMode.Open, FileAccess.Read))
+{
+    //Load the file stream into a Word document.
+    using (WordDocument document = new WordDocument(docStream, FormatType.Docx))
+    {
+        //Hook the event to customize the image. 
+        document.SaveOptions.ImageNodeVisited += SaveImage;
+        //Save a Markdown file to the MemoryStream.
+        MemoryStream outputStream = new MemoryStream();
+        document.Save(outputStream, FormatType.Markdown);
+        outputStream.Position = 0;
+        //Download as a Markdown file in the browser.
+        FileStream outStream = File.Create("WordToMd.md");
+        outputStream.CopyTo(outStream);
+        outputStream.Dispose();
+        outStream.Dispose();
+    }
+}
+static void SaveImage(object sender, ImageNodeVisitedEventArgs args)
+        {
+            string imagepath = @"D:\Temp\Image1.png";
+            //Save the image stream as file. 
+            using (FileStream fileStreamOutput = File.Create(imagepath))
+                args.ImageStream.CopyTo(fileStreamOutput);
+            //Set the uri to be used for image in output Markdown. 
+            args.Uri = imagepath;
+        }
+{% endhighlight %}
+
+{% highlight c# tabtitle="Xamarin" %}
+//Open the file as Stream.
+using (Stream docStream = typeof(App).GetTypeInfo().Assembly.GetManifestResourceStream("Sample.Assets.Input.docx"))
+{
+    //Load the file stream into a Word document.
+    using (WordDocument document = new WordDocument(docStream, FormatType.Docx))
+    {
+        //Hook the event to customize the image. 
+        document.SaveOptions.ImageNodeVisited += SaveImage;
+        //Save a Markdown file to the MemoryStream.
+        MemoryStream outputStream = new MemoryStream();
+        document.Save(outputStream, FormatType.Markdown);
+        //Save the stream as a file in the device and invoke it for viewing. 
+        Xamarin.Forms.DependencyService.Get<ISave>().SaveAndView("WordtoMd.md", "application/msword", outputStream);
+    }
+//Please download the helper files from the below link to save the stream as file and open the file for viewing in Xamarin platform
+//https://help.syncfusion.com/file-formats/docio/create-word-document-in-xamarin#helper-files-for-xamarin
+}
+static void SaveImage(object sender, ImageNodeVisitedEventArgs args)
+        {
+            string imagepath = @"D:\Temp\Image1.png";
+            //Save the image stream as file. 
+            using (FileStream fileStreamOutput = File.Create(imagepath))
+                args.ImageStream.CopyTo(fileStreamOutput);
+            //Set the uri to be used for image in output markdown. 
+            args.Uri = imagepath;
+        }
+{% endhighlight %}
+{% endtabs %}
+
+## Supported Word document elements
+
+The following table shows the list of supported elements while converting from Word document to Markdown.
```

Comment:
The following table shows the list of Word document elements supported in Word to Markdown conversion.